From 60a1b71f20d1d98c5446de68284d34455fa33f34 Mon Sep 17 00:00:00 2001
From: Tarun Bansal <tbansal@google.com>
Date: Fri, 21 Jul 2023 19:06:26 +0000
Subject: [PATCH] [M115] Return early if search text directive is empty

Bug: b:290675026, 1449021
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4678123
Reviewed-by: Jun Zou <junzou@chromium.org>
Commit-Queue: Tarun Bansal <tbansal@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1169437}
(cherry picked from commit 9745ab5b64c41cab0724653fa4a6a5c89f217750)

Change-Id: I20148cdd5ce2d2c616a40f513d457ba6586e4c7e
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4706982
Commit-Queue: Marco Georgaklis <mgeorgaklis@google.com>
Reviewed-by: Jun Zou <junzou@chromium.org>
Cr-Commit-Position: refs/branch-heads/5790@{#1796}
Cr-Branched-From: 1d71a337b1f6e707a13ae074dca1e2c34905eb9f-refs/heads/main@{#1148114}
---

diff --git a/chrome/browser/companion/text_finder/text_finder.cc b/chrome/browser/companion/text_finder/text_finder.cc
index 188b401..13fdacd 100644
--- a/chrome/browser/companion/text_finder/text_finder.cc
+++ b/chrome/browser/companion/text_finder/text_finder.cc
@@ -17,6 +17,7 @@
     FinishedCallback callback,
     AgentDisconnectHandler agent_disconnect_handler)
     : text_directive_(text_directive), receiver_(this) {
+  DCHECK(!text_directive.empty());
   InitializeAndBindToAnnotationAgent(agent_container, std::move(callback));
   SetAgentDisconnectHandler(std::move(agent_disconnect_handler));
 }
diff --git a/chrome/browser/companion/text_finder/text_finder_manager.cc b/chrome/browser/companion/text_finder/text_finder_manager.cc
index 9a8e5c2..bba3f56 100644
--- a/chrome/browser/companion/text_finder/text_finder_manager.cc
+++ b/chrome/browser/companion/text_finder/text_finder_manager.cc
@@ -26,9 +26,13 @@
 
 TextFinderManager::~TextFinderManager() = default;
 
-base::UnguessableToken TextFinderManager::CreateTextFinder(
+absl::optional<base::UnguessableToken> TextFinderManager::CreateTextFinder(
     const std::string& text_directive,
     TextFinder::FinishedCallback callback) {
+  if (text_directive.empty()) {
+    std::move(callback).Run(std::make_pair(text_directive, false));
+    return absl::nullopt;
+  }
   // Generate a unique random id.
   const base::UnguessableToken id = base::UnguessableToken::Create();
 
diff --git a/chrome/browser/companion/text_finder/text_finder_manager.h b/chrome/browser/companion/text_finder/text_finder_manager.h
index c1c8b73..e732928 100644
--- a/chrome/browser/companion/text_finder/text_finder_manager.h
+++ b/chrome/browser/companion/text_finder/text_finder_manager.h
@@ -34,7 +34,7 @@
   // Calls `callback` with the boolean search result indicating found or not
   // found. Removes the finder from manager upon finishing searching or agent
   // disconnection. Returns the id associated with the created text finder.
-  base::UnguessableToken CreateTextFinder(
+  absl::optional<base::UnguessableToken> CreateTextFinder(
       const std::string& text_directive,
       TextFinder::FinishedCallback callback);
 
diff --git a/chrome/browser/companion/text_finder/text_finder_manager_unittest.cc b/chrome/browser/companion/text_finder/text_finder_manager_unittest.cc
index 4526d55..ddb656b 100644
--- a/chrome/browser/companion/text_finder/text_finder_manager_unittest.cc
+++ b/chrome/browser/companion/text_finder/text_finder_manager_unittest.cc
@@ -37,10 +37,11 @@
           [](std::pair<std::string, bool> text_found) { return; });
   const auto id = manager->CreateTextFinder(
       text_directive, std::move(finished_finding_callback));
+  EXPECT_TRUE(id.has_value());
   EXPECT_EQ(manager->Size(), 1u);
 
   // Remove text finder.
-  manager->RemoveTextFinder(id);
+  manager->RemoveTextFinder(id.value());
   EXPECT_EQ(manager->Size(), 0u);
 }
 
diff --git a/chrome/browser/companion/text_finder/text_highlighter.cc b/chrome/browser/companion/text_finder/text_highlighter.cc
index 1973468..1ad7bab 100644
--- a/chrome/browser/companion/text_finder/text_highlighter.cc
+++ b/chrome/browser/companion/text_finder/text_highlighter.cc
@@ -17,6 +17,7 @@
     const std::string& text_directive,
     const mojo::Remote<blink::mojom::AnnotationAgentContainer>& agent_container)
     : text_directive_(text_directive), receiver_(this) {
+  DCHECK(!text_directive.empty());
   InitializeAndBindToAnnotationAgent(agent_container);
 }
 
